src/drivers/ins/microstrain/MicroStrain.cpp

```diff
@@ -338,27 +511,283 @@ mip_cmd_result MicroStrain::configureImuMessageFormat()
+	}
+
+	// Write the settings
+	res = writeMessageFormat(MIP_FILTER_DATA_DESC_SET, num_filter_descriptors,
+				 filter_descriptors);
+
+	return res;
+}
+
+mip_cmd_result MicroStrain::configureAidingMeasurement(uint16_t aiding_source, bool enable)
+{
+	mip_cmd_result res;
+
+	// Configures the aiding measurement if the device support it
+	if (supportsDescriptor(MIP_FILTER_CMD_DESC_SET, MIP_CMD_DESC_FILTER_AIDING_MEASUREMENT_ENABLE)) {
+		res = mip_filter_write_aiding_measurement_enable(&_device, aiding_source, enable);
+
+		// If the device doesnt support the aiding source but the command is to disable it, we consider it a success
+		if (res == MIP_NACK_INVALID_PARAM && !enable) {
+			res = MIP_ACK_OK;
+		}
+
+	} else {
+		PX4_WARN("Aiding measurements are not supported");
+
+		// If the command was to disable, we consider it a success
+		if (!enable) {
+			res = MIP_ACK_OK;
+
+		} else {
+			res = MIP_PX4_ERROR;
+		}
+	}
+
+	return res;
+}
+
+mip_cmd_result MicroStrain::configureAidingSources()
+{
+	PX4_DEBUG("Configuring aiding sources");
+	mip_cmd_result res;
+
+	// Selectively turn on the magnetometer aiding source
+	if (!mip_cmd_result_is_ack(res = configureAidingMeasurement(
+			MIP_FILTER_AIDING_MEASUREMENT_ENABLE_COMMAND_AIDING_SOURCE_MAGNETOMETER,
+			_param_ms_int_mag_en.get()))) {
+		PX4_ERR("Could not configure magnetometer aiding");
+		return res;
+	}
+
+	// Enables GNSS Position & Velocity as an aiding measurement
+	res = configureAidingMeasurement(MIP_FILTER_AIDING_MEASUREMENT_ENABLE_COMMAND_AIDING_SOURCE_GNSS_POS_VEL,
+					 true);
+
+	if (!mip_cmd_result_is_ack(res)) {
+		if (res != MIP_NACK_INVALID_PARAM && res != MIP_PX4_ERROR) {
+			PX4_ERR("Error enabling GNSS Position & Velocity aiding");
+			return res;
+
+		} else {
+			PX4_WARN("Could not enable GNSS Position & Velocity aiding");
+		}
+
+	} else {
+		// Check to see if sending GNSS position and velocity as an aiding measurement is supported
+		bool pos_aiding = supportsDescriptor(MIP_AIDING_CMD_DESC_SET, MIP_CMD_DESC_AIDING_POS_LLH);
+		bool vel_aiding = supportsDescriptor(MIP_AIDING_CMD_DESC_SET, MIP_CMD_DESC_AIDING_VEL_NED);
+		_ext_pos_vel_aiding = pos_aiding && vel_aiding;
+
+		if (!_ext_pos_vel_aiding) {
+			PX4_ERR("Sending GNSS pos/vel aiding messages is not supported");
+			return MIP_PX4_ERROR;
+		}
+
+	}
+
+	// Selectively turn on external heading as an aiding measurement
+	if (!mip_cmd_result_is_ack(res = configureAidingMeasurement(
+			MIP_FILTER_AIDING_MEASUREMENT_ENABLE_COMMAND_AIDING_SOURCE_GNSS_HEADING,
+			_param_ms_ext_heading_en.get()))) {
+		PX4_ERR("Could not configure external heading aiding");
+		return res;
+
+	} else {
+		// Check to see if sending external heading as an aiding measurement is supported
+		_ext_heading_aiding = supportsDescriptor(MIP_AIDING_CMD_DESC_SET, MIP_CMD_DESC_AIDING_HEADING_TRUE)
+				      && _param_ms_ext_heading_en.get();
+
+		if (!_ext_heading_aiding && _param_ms_ext_heading_en.get()) {
```

Comment:
Same here. Is this a failure state? This disqualifies some devices I am not sure if we want to disqualify